include/event2/dns.h

```diff
@@ -659,10 +664,48 @@ typedef void (*evdns_request_callback_fn_type)(struct evdns_server_request *, vo
  */
 EVENT2_EXPORT_SYMBOL
 struct evdns_server_port *evdns_add_server_port_with_base(struct event_base *base, evutil_socket_t socket, int flags, evdns_request_callback_fn_type callback, void *user_data);
+
+struct evconnlistener;
+
+/** Create a new TCP DNS server port.
+
+    @param base The event base to handle events for the server port.
+    @param listener A TCP listener to accept DNS requests.
+    @param flags Always 0 for now.
+    @param callback A function to invoke whenever we get a DNS request
+      on the socket.
+    @param user_data Data to pass to the callback.
+    @return an evdns_server_port structure for this server port or NULL if
+      an error occurred.
+ */
+EVENT2_EXPORT_SYMBOL
+struct evdns_server_port *evdns_add_server_port_with_listener(
+    struct event_base *base, struct evconnlistener *listener, int flags,
+    evdns_request_callback_fn_type callback, void *user_data);
+
 /** Close down a DNS server port, and free associated structures. */
 EVENT2_EXPORT_SYMBOL
 void evdns_close_server_port(struct evdns_server_port *port);
 
+/**
+  Set the value of a configuration option for the DNS server.
+
+  The currently available configuration options are:
+
+  - max-tcp-clients - maximum number of simultaneous tcp connections
+	(clients) that server can hold (default value = 10). Can be set
+	only for TCP DNS server.
+
+  Option names are allowed to end with colon.
```

Comment:
Allow colon at the end of the option name for consistency with `evdns_base_set_option`. Not 100 % sure it is good.